pkg/ingester/ingester.go

```diff
@@ -211,6 +251,49 @@ func New(cfg Config, clientConfig client.Config, limits *validation.Overrides, c
 	return i, nil
 }
 
+// NewV2 returns a new Ingester that uses prometheus block storage instead of chunk storage
+func NewV2(cfg Config, clientConfig client.Config, limits *validation.Overrides, chunkStore ChunkStore, registerer prometheus.Registerer) (*Ingester, error) {
+
+	var bkt *s3.Bucket
+	s3Cfg := s3.Config{
+		Bucket:    cfg.S3Bucket,
```

Comment:
This can definitely hit the same API limitation of S3 providers (and has in my testing). I'm currently working on a follow-on PR to shard across buckets as in #1625. It's mostly straight forward except for the `SyncBlocks` func which iterates over a bucket.